lib/client.dart

```diff
@@ -460,27 +459,35 @@ class Router {
    * value is null which then determines the behavior based on
    * [History.supportsState].
    */
-  Router({bool useFragment, Window windowImpl, bool sortRoutes: true,
-         RouterLinkMatcher linkMatcher, WindowClickHandler clickHandler})
-      : this._init(null, useFragment: useFragment, windowImpl: windowImpl,
-          sortRoutes: sortRoutes, linkMatcher: linkMatcher,
-          clickHandler: clickHandler);
-
-
-  Router._init(Router parent, {bool useFragment, Window windowImpl,
-      this.sortRoutes, RouterLinkMatcher linkMatcher,
+  Router(
+      {bool useFragment,
+      HistoryProvider historyProvider,
```

Comment:
TODO - `useFragment` and `historyProvider` have blurred responsibilities at the moment
- `useFragment` == true implies `HashProvider`
- `useFragment` == false implies `BrowserProvider`

consider refactoring this gray area away